core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -837,6 +845,18 @@ public MergeWriterNode plan(Merge merge)
             }
 
             whenClauses.add(new WhenClause(condition, new Row(rowBuilder.build())));
+
+            if (!constraints.isEmpty()) {
+                checkConstraintsAssignments.putIdentity(uniqueIdSymbol);
+                checkConstraintsAssignments.putIdentity(presentColumn);
+                checkConstraintsAssignments.putIdentity(rowIdSymbol);
+                checkConstraintsAssignments.putIdentities(source.getFieldMappings());
```

Comment:
A few lines below not all the field mappings seem to be used 

https://github.com/trinodb/trino/blob/77dc3fd6b534058f88e9e8714936c21c2502b228/core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java#L856-L862

Is there a specific reason why you chose to use them all here?

Reply:
Test cases fails if we don't add all those identities. Especially some complicated queries like for example using symbols from source table in `INSERT` statement:
```
MERGE INTO mock.tiny.nation USING (VALUES (26, 'POLAND', 10, 'No comment'), (27, 'HOLLAND', 10, 'A comment')) t(a,b,c,d) ON nationkey = a
                WHEN NOT MATCHED THEN INSERT VALUES (a,b,c,d)
```